supervisord.py

```diff
@@ -47,14 +49,27 @@ def check(self, instance):
             AgentCheck.UNKNOWN: 0
         }
 
-        # Report service checks and uptime for each process
-        proc_names = instance.get('proc_names')
-        if proc_names and len(proc_names):
-            processes = [server.supervisor.getProcessInfo(p) for p in proc_names]
-        else:
-            processes = server.supervisor.getAllProcessInfo()
+        processes = server.supervisor.getAllProcessInfo()
+        proc_regex = instance.get('proc_regex', [])
+        proc_names = instance.get('proc_names', [])
+        monitored_processes = []
+
+        if not len(proc_regex) and not len(proc_names):
+            monitored_processes = processes
+
+        if len(proc_regex):
```

Comment:
You also don't need these anymore; they will just be no-ops if the lists are empty.